### PR TITLE
[1LP][RFR] StackCollection + tests

### PR DESCRIPTION
**Pull request by vatsalparekh**
Purpose or Intent
=================

Collection implementation for Stack + its tests 
{{pytest: cfme/tests/cloud/test_stack.py -vvvv}}

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit 8187e152ce6e87882df5ced902b9ffc270ecdf6f:

`cfme/cloud/stack.py`:
- :bangbang: Line 261:32: **E231** *missing whitespace after ','*
- :bangbang: Line 275:101: **E501** *line too long (113 > 100 characters)*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

File: cfme/cloud/stack.py [hunk 1]
```diff
@@ -263,41 +296,20 @@ def __init__(self, name, provider, quad_name=None, appliance=None):
...
-        elif from_dest == 'Details':
-            view.toolbar.configuration.item_select('Remove this Orchestration Stack',
-                                                   handle_alert=True)
-            view.entities.flash.assert_no_error()
-            view.entities.flash.assert_success_message('Delete initiated for 1 Orchestration Stack')
+    def delete(self):
+        """Delete the stack from detail view"""
+        view = navigate_to(self, 'Details')
+        view.toolbar.configuration.item_select('Remove this Orchestration Stack', handle_alert=True)
+        view.entities.flash.assert_no_error()
```
Comment (quarckster) [inline cfme/cloud/stack.py]: **REQUIRED**
This line is not needed

Comment (vatsalparekh) [inline cfme/cloud/stack.py]: Removed

File: cfme/cloud/stack.py [hunk 2]
```diff
@@ -379,7 +394,7 @@ def resetter(self):
 @navigator.register(Stack, 'Details')
 class Details(CFMENavigateStep):
     VIEW = StackDetailsView
-    prerequisite = NavigateToSibling('All')
+    prerequisite = NavigateToAttribute('collection', 'All')
```
Comment (quarckster) [inline cfme/cloud/stack.py]: Is it working? What is 'collection'?

Comment (vatsalparekh) [inline cfme/cloud/stack.py]: Yes it works, `collection` is the reference to the [Collection](https://github.com/vatsalparekh/integration_tests/blob/27e4383d04b436057b9f4e8fea635793556d6785/cfme/cloud/stack.py#L298)
 object/class

File: cfme/cloud/stack.py [hunk 3]
```diff
@@ -327,7 +342,7 @@ def get_tags(self):
 
     def wait_for_exists(self):
         """Wait for the row to show up"""
-        view = navigate_to(self, 'All')
+        view = navigate_to(StackCollection, 'All')
```
Comment (mfalesni) [inline cfme/cloud/stack.py]: `navigate_to(self.collection, 'All')` should be future-proof.

The sense of collections is to remove registering and using just classes without context.

Comment (vatsalparekh) [inline cfme/cloud/stack.py]: Changed.

**Pull request merged**